test/TopupEUReTest.t.sol

```diff
@@ -66,29 +66,33 @@ contract TopupTest is BaseFixture {
         IVault.ExitPoolRequest memory request =
             abi.decode(abi.decode(entries[1].data, (bytes)), (IVault.ExitPoolRequest));
 
-        bytes memory eventPayloadGenerated = abi.encodeWithSelector(
-            IVault.exitPool.selector, roboModule.BPT_STEUR_EURE_POOL_ID(), GNOSIS_SAFE, payable(GNOSIS_SAFE), request
+        _assertPreStorageValuesNextTxExec(
+            address(roboModule.BALANCER_VAULT()),
+            abi.encodeWithSelector(
+                IVault.exitPool.selector,
+                roboModule.BPT_STEUR_EURE_POOL_ID(),
+                GNOSIS_SAFE,
+                payable(GNOSIS_SAFE),
+                request
+            )
         );
 
-        _assertPreStorageValuesNextTxExec(address(roboModule.BALANCER_VAULT()), eventPayloadGenerated);
-
         vm.prank(KEEPER);
         roboModule.adjustPool(RoboSaverVirtualModule.PoolAction.EXEC_QUEUE_POOL_ACTION, 0);
 
-        // (uint256 nonce, address target, bytes memory execTxPayload) = roboModule.txQueueData();
-        // delayModule.executeNextTx(address(roboModule.BALANCER_VAULT()), 0, execTxPayload, Enum.Operation.Call);
-
         // ensure default values at `txQueueData` after execution
         _assertPostDefaultValuesNextTxExec();
 
-        assertLt(
+        assertApproxEqAbs(
             IERC20(BPT_STEUR_EURE).balanceOf(GNOSIS_SAFE),
-            initialBptBal,
+            initialBptBal - maxBPTAmountIn,
+            DIFF_MIN_OUT_CALC_ALLOWED,
             "BPT balance: not decreased after withdrawing from the pool"
         );
-        assertGt(
+
+        assertEq(
             IERC20(EURE).balanceOf(GNOSIS_SAFE),
-            initialEureBal,
+            initialEureBal + _deficit,
             "EURE balance: not increased after withdrawing from the pool"
```

Comment:
also not accurate description?

Reply:
ad536a7b16c5679d767dd165cabe610577ad87f4